vowpalwabbit/v_array.h

```diff
@@ -31,6 +31,21 @@ const size_t erase_point = ~((1u << 10u) - 1u);
 template <class T>
 struct v_array
 {
+private:
+  void delete_v_array()
+  {
+    if (_begin != nullptr)
+    {
+      for (iterator item = _begin; item != _end; ++item) { item->~T(); }
```

Comment:
The rest of v_array assumes it only contains trivial types, do we want to maintain that assumption (for now)? We can add non-trivial type support in one shot later

Reply:
There are places where it does not contain trivial types at the moment, such as decision_scores. So I think we need this